### PR TITLE
feat: Accessibility screen on profile page

### PR DESCRIPTION
**Pull request by reidzeibel**
closes: https://github.com/AtB-AS/kundevendt/issues/17887

Depends on: https://github.com/AtB-AS/mittatb-app/pull/4760, https://github.com/AtB-AS/mittatb-app/pull/4761 https://github.com/AtB-AS/config-specs/pull/59 https://github.com/AtB-AS/firestore-configuration/pull/678

In this PR: 
- "Accessibility" option on profile menu, and screen implementation.
- Firestore texts added (need to check translation and styling).
- Added remote config for kundeservice phone number.
- Implement preference for travel aid toggle

### CODE REVIEW

**Comment by gorandalum** [comment]
I just skimmed through this. Remember to split it into logical pull requests. Changes to shared components like  `ToggleSectionItem` and `ThemeText` deserves their own PR.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_RootScreen.tsx [hunk 1]
```diff
@@ -46,14 +46,15 @@ import {ContentHeading} from '@atb/components/heading';
...
+import { useIsTravelAidEnabled } from '@atb/travel-aid/use-is-travel-aid-enabled';
 
 const buildNumber = getBuildNumber();
 const version = getVersion();
 
 type ProfileProps = ProfileScreenProps<'Profile_RootScreen'>;
 
 export const Profile_RootScreen = ({navigation}: ProfileProps) => {
-  const {enable_ticketing} = useRemoteConfig();
+  const {enable_ticketing, enable_vipps_login} = useRemoteConfig();
```
Comment (jorelosorio) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_RootScreen.tsx:57]: We do nothing with `enable_vipps_login` it was just moved here. Before there were two `useRemoteConfig` usages for different atrrs.

File: src/configuration/FirestoreConfigurationContext.tsx [hunk 2]
```diff
@@ -43,8 +43,16 @@ import {useResubscribeToggle} from '@atb/utils/use-resubscribe-toggle';
 
 export const defaultVatPercent: number = 12;
 
+export type AppTextKey =
+  | 'travelAidSubText'
+  | 'travelAidHowToTitle'
+  | 'travelAidHowToContent'
+  | 'travelAidKeepInMindTitle'
+  | 'travelAidKeepInMindContent'
```
Comment (gorandalum) [inline src/configuration/FirestoreConfigurationContext.tsx:51]: I would advice against having all of these in Firestore. Lets talk in the office tomorrow, also with designer.

File: src/remote-config.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import remoteConfig from '@react-native-firebase/remote-config';
 import {ENABLE_TICKETING, PRIVACY_POLICY_URL, CUSTOMER_SERVICE_URL} from '@env';
 
 export type RemoteConfig = {
+  contact_phone_number: string;
```
Comment (gorandalum) [inline src/remote-config.ts:5]: We have used Firestore lately for things that doesn't need the capabilities of Remote Config, for example for URLs, and I think phone number is pretty much the same? Or do we envision to have some conditions on the phone number in Remote Config?

Comment (jorelosorio) [reply src/remote-config.ts:5]: It was decided at meetings to have this kind of information stored in the remote config to edit as if something goes wrong, The same applies to the phone, if we set it empty in the remote config then the button disappears. We need to involve the product owner and ask him, but that is what we decided on meetings.

File: src/configuration/FirestoreConfigurationContext.tsx (hunk 2 above)
Comment (jorelosorio) [reply src/configuration/FirestoreConfigurationContext.tsx:51]: The project owner needs to be involved in decisions like removing this from the remote config, it was decided on meetings.

File: src/remote-config.ts (hunk 3 above)
Comment (gorandalum) [reply src/remote-config.ts:5]: I don't think a non-technical product owner is the one that decides between having a phone number in RemoteConfig vs Firestore. The product owner states the need, and it is the developers that knows if Remote Config or Firestore is fit for the job.

If this is the need:
- Able to specify phone number outside of app release
- If no phone number specified, don't show button

… then I think Firestore is the correct place for it.

File: src/configuration/FirestoreConfigurationContext.tsx (hunk 2 above)
Comment (gorandalum) [reply src/configuration/FirestoreConfigurationContext.tsx:51]: I created a Slack thread for this question:
https://mittatb.slack.com/archives/C02EEG7D8EL/p1729151816404169

File: src/remote-config.ts (hunk 3 above)
Comment (jorelosorio) [reply src/remote-config.ts:5]: The decision was not taken in a technical matter, it was based on the need to perhaps change texts or functionality as the first phase changes or requires actions.

File: src/preferences/types.ts [hunk 4]
```diff
@@ -24,6 +24,9 @@ export type UserPreferences = {
   showTestIds?: boolean;
   hideProductDescriptions?: boolean;
   debugShowSeconds?: boolean;
+  // A toggle inside the accessibility settings: it only disables the travel aid feature in the app,
+  // not the entire accessibility settings
+  travelAid?: boolean;
```
Comment (rosvik) [inline src/preferences/types.ts:29]: We should make naming consistent here it a separate PR ("journey aid" seems to be correct), but let's make the local storage stuff correct from the start:

```suggestion
  journeyAidEnabled?: boolean;
```

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      'Journey Aid is a simplified mode designed specifically for those who need extra support when traveling.\n\n* Use the app to signal a stop for a bus or tram from your boarding stop.\n* Get faster access to real-time updates on arrivals and upcoming stops, as well as a shortcut to your tickets.* Shortcut to valid tickets.* You can request an invitation code for this feature through the The Norwegian Association of the Blind and Partially Sighted or AtB’s customer service.',
+      'Reisehjelp er en forenklet modus spesielt tilrettelagt for deg som trenger ekstra støtte når du reiser.\n\n* Stoppe en buss eller trikk med appen, fra holdeplassen der du skal på.\n* Raskere tilgang på sanntids-oppdateringer om ankomst og neste holdeplass.\n* Snarvei til gyldige billetter.\n* Invitasjonskode for tilgang til funksjonen får du gjennom Blindeforbundet eller AtBs kundeservice.',
```
Comment (rosvik) [inline src/translations/screens/subscreens/TravelAidSettingsTexts.ts:18]:
```suggestion
      'Reisehjelp er en forenklet modus spesielt tilrettelagt for deg som trenger ekstra støtte når du reiser.\n\n* Stopp en buss eller trikk med appen, fra holdeplassen der du skal på.\n* Raskere tilgang på sanntids-oppdateringer om ankomst og neste holdeplass.\n* Snarvei til gyldige billetter.',
      'Journey Aid is a simplified mode designed specifically for those who need extra support when traveling.\n\n* Use the app to signal a stop for a bus or tram from your boarding stop.\n* Get faster access to real-time updates on arrivals and upcoming stops, as well as a shortcut to your tickets.* Shortcut to valid tickets.',
      'Reisehjelp er ein forenkla modus spesielt tilrettelagt for deg som treng ekstra støtte når du reiser.\n\n* Stopp ein buss eller trikk med appen, frå haldeplassen der du skal på.\n* Raskare tilgang på sanntids-oppdateringar om ankomst og neste haldeplass.\n* Snarveg til gyldige billettar.',
```

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    howTo: {
+      title: _(
+        'Hvordan sende stopp',
+        'How to send a stop',
+        'Hvordan sende stopp',
+      ),
+      content: _(
+        'Stoppsignal fra appen varsler sjåfør om at noen med ekstra behov skal av eller på.\n\n<br/>* I “Avganger”-fanen, velg holdeplassen du skal reise fra.\n* Fra buss-listen, velg riktig buss/trikk.\n* I visningen for bussen/trikken, trykk på Reisehjelp.\n* I Reisehjelp-modusen, trykk “Send stopp”.',
+        '',
+        '',
```
Comment (rosvik) [inline src/translations/screens/subscreens/TravelAidSettingsTexts.ts:55]: Need to add translations here

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      ),
+    },
+    keepInMind: {
+      title: _('Husk:', 'Keep in mind:', 'Husk:'),
+      content: _(
+        '* Følg retningslinjer, som å stå på ledelinjen eller lengst frem på holdeplassen, for å vise at du skal på.\n* Du kan bare stoppe én buss/ trikk om gangen.\n* Du kan ikke angre et stoppsignal som er sendt.',
+        '* Follow guidelines, such as standing on the tactile guiding line or at the front of the stop, to signal that you’re boarding.\n* You can only stop one bus/tram at a time.\n * You cannot cancel a stop signal once it has been sent.',
+        '',
```
Comment (rosvik) [inline src/translations/screens/subscreens/TravelAidSettingsTexts.ts:63]:
```suggestion
        '* Følg retningslinjer, som å stå på ledelinjen eller lengst frem på holdeplassen, for å vise at du skal på.\n* Du kan bare stoppe én buss/trikk om gangen.\n* Du kan ikke angre et stoppsignal som er sendt.',
        '* Follow guidelines, such as standing on the tactile guiding line or at the front of the stop, to signal that you’re boarding.\n* You can only stop one bus/tram at a time.\n * You cannot cancel a stop signal once it has been sent.',
        '* Følg retningslinjer, som å stå på ledelinja eller lengst fram på haldeplassen, for å vise at du skal på.\n* Du kan berre stoppe ein buss/trikk om gongen.\n* Du kan ikkje angre eit stoppsignal som er sendt.',
```

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_RootScreen.tsx [hunk 8]
```diff
@@ -275,6 +276,20 @@ export const Profile_RootScreen = ({navigation}: ProfileProps) => {
 
           <ContentHeading text={t(ProfileTexts.sections.settings.heading)} />
           <Section>
+
+          {travelAidEnabled? (
```
Comment (rosvik) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_RootScreen.tsx:280]:
```suggestion
          {travelAidEnabled ? (
```

**Review by gorandalum — CHANGES_REQUESTED** [review]
Looks good! Only thing left is to use Firestore instead of RemoteConfig for phone number if no specific arguments for using RemoteConfig occur. Firestore is generally preferable because of its integration with GitHub and config-specs.

**Comment by jorelosorio** [comment]
> Looks good! Only thing left is to use Firestore instead of RemoteConfig for phone number if no specific arguments for using RemoteConfig occur. Firestore is generally preferable because of its integration with GitHub and config-specs.

On meetings, we decided that it is ok to have in remote config because changes are not required immediately.

**Comment by rosvik** [comment]
@jorelosorio If I remember it correctly, we mentioned that it was OK to have the 12 hour delay with remote config, but that doesn't mean that using firestore is wrong / less good. I think adding it to firestore under [`Other`](https://github.com/AtB-AS/config-specs/blob/main/src/other.ts) is a good solution.

**Comment by jorelosorio** [comment]
> @jorelosorio If I remember it correctly, we mentioned that it was OK to have the 12 hour delay with remote config, but that doesn't mean that using firestore is wrong / less good. I think adding it to firestore under [`Other`](https://github.com/AtB-AS/config-specs/blob/main/src/other.ts) is a good solution.

It is not wrong or less good but was decided in that way.

**Comment by jorelosorio** [comment]
> Looks good! Only thing left is to use Firestore instead of RemoteConfig for phone number if no specific arguments for using RemoteConfig occur. Firestore is generally preferable because of its integration with GitHub and config-specs.

@gorandalum where do you think in the firebase structure, it feels safe to see `contact_phone_number`?

**Comment by jorelosorio** [comment]
> > Looks good! Only thing left is to use Firestore instead of RemoteConfig for phone number if no specific arguments for using RemoteConfig occur. Firestore is generally preferable because of its integration with GitHub and config-specs.
> 
> @gorandalum, where do you think it feels safe to see `contact_phone_number` in the firebase structure?

I did not hear suggestions, I added it to others.

**Comment by jorelosorio** [comment]
> Looks good! Only thing left is to use Firestore instead of RemoteConfig for phone number if no specific arguments for using RemoteConfig occur. Firestore is generally preferable because of its integration with GitHub and config-specs.

@gorandalum changes were made, as you asked. It did not take 10m but implementation is done in the 3 repositories for approval.

File: src/configuration/FirestoreConfigurationContext.tsx [hunk 9]
```diff
@@ -152,6 +156,12 @@ export const FirestoreConfigurationContextProvider: React.FC = ({children}) => {
             setVatPercent(vatPercent);
           }
 
+          const contactPhoneNumber =
+            getContactPhoneNumberFromSnapshot(snapshot);
+          if (contactPhoneNumber) {
+            setContactPhoneNumber(contactPhoneNumber);
+          }
+
```
Comment (gorandalum) [inline src/configuration/FirestoreConfigurationContext.tsx:164]: Should we `setContactPhoneNumber` even if the value is `undefined`?

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_TravelAidScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+                  accessibilityRole="button"
+                  testID="travelAidContactCustomerServiceButton"
+                  onPress={async () => {
+                    const phoneNumber = `tel:${contactPhoneNumber}`;
+                    if (await Linking.canOpenURL(phoneNumber)) {
+                      Linking.openURL(phoneNumber);
+                    } else {
+                      Bugsnag.notify(
+                        new Error(
+                          'Could not open phone number in accessiblity settings',
```
Comment (gorandalum) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_TravelAidScreen.tsx:96]: Could add the phone number to the error msg here, so we see it if there is something wrong with our logic and the link can't be opened because the phone number is invalid.

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+      title: _('Viktig å vite', 'Important information', 'Viktig å vite'),
+    },
+  },
+
+  toggle: {
+    title: _('Reisehjelp', 'Journey Aid', 'Reisehjelp'),
+    subText: _(
+      'Reisehjelp er en forenklet modus spesielt tilrettelagt for deg som trenger ekstra støtte når du reiser.\n\n* Stopp en buss eller trikk med appen, fra holdeplassen der du skal på.\n* Raskere tilgang på sanntids-oppdateringer om ankomst og neste holdeplass.\n* Snarvei til gyldige billetter.',
+      'Journey Aid is a simplified mode designed specifically for those who need extra support when traveling.\n\n* Use the app to signal a stop for a bus or tram from your boarding stop.\n* Get faster access to real-time updates on arrivals and upcoming stops, as well as a shortcut to your tickets.* Shortcut to valid tickets.',
+      'Reisehjelp er ein forenkla modus spesielt tilrettelagt for deg som treng ekstra støtte når du reiser.\n\n* Stopp ein buss eller trikk med appen, frå haldeplassen der du skal på.\n* Raskare tilgang på sanntids-oppdateringar om ankomst og neste haldeplass.\n* Snarveg til gyldige billettar.',
```
Comment (gorandalum) [inline src/translations/screens/subscreens/TravelAidSettingsTexts.ts:18]: I'm a little hesitant about adding these texts in this PR. The texts isn't in sync with what features we have in the app now, and it mentions more than one of them. For example both blue button, and shortcut to my tickets, which are two features that may be added in different phases.

I think it would be more safe to always have the text in the app in sync with the features in the app. Adding texts doesn't take much time, so there is no real benefit to pre-adding text about features that will come in the future.

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+        'Hvordan sende stopp',
+        'How to send a stop',
+        'Hvordan sende stopp',
+      ),
+      content: _(
+        'Stoppsignal fra appen varsler sjåfør om at noen med ekstra behov skal av eller på.\n\n* I “Avganger”-fanen, velg holdeplassen du skal reise fra.\n* Fra buss-listen, velg riktig buss/trikk.\n* I visningen for bussen/trikken, trykk på Reisehjelp.\n* I Reisehjelp-modusen, trykk “Send stopp”.',
+        'A stop signal from the app alerts the driver that someone with extra needs wants to get off or on.\n\n* In the "Departures" tab, select the stop you are departing from.\n* From the bus list, choose the correct bus/tram.\n* In the view for the bus/tram, tap on "Travel Assistance."\n* In Travel Assistance mode, press "Send Stop."',
+        'Stoppsignal frå appen varslar sjåføren om at nokon med ekstra behov skal av eller på.\n\n* I "Avganger"-fanen, vel haldeplassen du skal reise frå.\n* Frå busslista, vel rett buss/trikk.\n* I visninga for bussen/trikken, trykk på "Reisehjelp."\n* I Reisehjelp-modusen, trykk "Send stopp."',
+      ),
+    },
```
Comment (gorandalum) [inline src/translations/screens/subscreens/TravelAidSettingsTexts.ts:57]: Same here regarding texts in app should be in sync with features in the app.

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    keepInMind: {
+      title: _('Husk:', 'Keep in mind:', 'Husk:'),
+      content: _(
+        '* Følg retningslinjer, som å stå på ledelinjen eller lengst frem på holdeplassen, for å vise at du skal på.\n* Du kan bare stoppe én buss/trikk om gangen.\n* Du kan ikke angre et stoppsignal som er sendt.',
+        '* Follow guidelines, such as standing on the tactile guiding line or at the front of the stop, to signal that you\'re boarding.\n* You can only stop one bus/tram at a time.\n * You cannot cancel a stop signal once it has been sent.',
+        '* Følg retningslinjer, som å stå på ledelinja eller lengst fram på haldeplassen, for å vise at du skal på.\n* Du kan berre stoppe ein buss/trikk om gongen.\n* Du kan ikkje angre eit stoppsignal som er sendt.',
```
Comment (gorandalum) [inline src/translations/screens/subscreens/TravelAidSettingsTexts.ts:63]: Same here about texts. We business rules of sending stops, like if more than one can be sent in parallell, isn't really decided yet, so I think it is premature to have it in the app.

**Review by gorandalum — CHANGES_REQUESTED** [review]

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+      'Journey Aid is a simplified view of bus and tram departures, designed for those who need extra support when traveling. It provides quick access to real-time updates on arrival times and the next stop.\n\n* Activate Journey Aid by tapping the switch at the top right.\n* Select your stop under the "Departures" tab.\n* Choose the correct bus or tram from the list.\n* Tap the Journey Aid button at the top of the screen.\n\nWe are still working on this feature, and new functionality will be added continuously. Please share your feedback with us!',
+      'Reisehjelp er en forenklet visning av buss- og trikkeavganger, tilpasset for deg som trenger ekstra støtte på reisen. Den gir rask tilgang til sanntidsoppdateringer om ankomsttid og neste holdeplass.\n\n* Aktiver Reisehjelp ved å trykke på bryteren øverst til høyre.\n* Velg holdeplass under «Avganger»-fanen.\n* Velg riktig buss eller trikk fra listen.\n* Trykk på knappen for Reisehjelp øverst i visningen.\n\nVi jobber fortsatt med funksjonen, og ny funksjonalitet vil komme fortløpende. Del gjerne din tilbakemelding!',
```
Comment (hanne-at-atb) [inline src/translations/screens/subscreens/TravelAidSettingsTexts.ts:18]:
```suggestion
      'Reisehjelp er ei forenkla visning av buss- og trikkeavgangar, tilpassa for deg som trenger ekstra støtte på reisa. Den gir rask tilgang til sanntidsoppdateringer om framkomsttid og neste haldeplass.\n\n* Aktivér Reisehjelp ved å trykke på brytaren øvst til høgre.\n* Velg haldeplass under «Avgangar»-fanen.\n* Velg riktig buss eller trikk frå lista.\n* Trykk på knappen for Reisehjelp øvst i visninga.\n\nVi jobbar fortsatt med funksjonen, og ny funksjonalitet vil komme fortløpande. Del gjerne din tilbakemelding!,
```

Comment (hanne-at-atb) [reply src/translations/screens/subscreens/TravelAidSettingsTexts.ts:18]: My best attempt on nynorsk translation.

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts (hunk 13 above)
Comment (jorelosorio) [reply src/translations/screens/subscreens/TravelAidSettingsTexts.ts:63]: This was removed now for this version,

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts (hunk 12 above)
Comment (jorelosorio) [reply src/translations/screens/subscreens/TravelAidSettingsTexts.ts:57]: Is not to be shown for this version.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_TravelAidScreen.tsx (hunk 10 above)
Comment (jorelosorio) [reply src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_TravelAidScreen.tsx:80]: Added.

File: src/translations/screens/subscreens/TravelAidSettingsTexts.ts [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+      'Journey Aid is a simplified view of bus and tram departures, designed for those who need extra support when traveling. It provides quick access to real-time updates on arrival times and the next stop.\n\n* Activate Journey Aid by tapping the switch at the top right.\n* Select your stop under the "Departures" tab.\n* Choose the correct bus or tram from the list.\n* Tap the Journey Aid button at the top of the screen.\n\nWe are still working on this feature, and new functionality will be added continuously. Please share your feedback with us!',
+      'Reisehjelp er ei forenkla visning av buss- og trikkeavgangar, tilpassa for deg som trenger ekstra støtte på reisa. Den gir rask tilgang til sanntidsoppdateringer om framkomsttid og neste haldeplass.\n\n* Aktivér Reisehjelp ved å trykke på brytaren øvst til høgre.\n* Velg haldeplass under «Avgangar»-fanen.\n* Velg riktig buss eller trikk frå lista.\n* Trykk på knappen for Reisehjelp øvst i visninga.\n\nVi jobbar fortsatt med funksjonen, og ny funksjonalitet vil komme fortløpande. Del gjerne din tilbakemelding!,
```
Comment (rosvik) [inline src/translations/screens/subscreens/TravelAidSettingsTexts.ts:18]:
```suggestion
      'Reisehjelp er ei forenkla visning av buss- og trikkeavgangar, tilpassa for deg som treng ekstra støtte på reisa. Den gir rask tilgang til sanntidsoppdateringar om framkomsttid og neste haldeplass.\n\n* Aktivér Reisehjelp ved å trykke på brytaren øvst til høgre.\n* Velg haldeplass under «Avgangar»-fanen.\n* Vel riktig buss eller trikk frå lista.\n* Trykk på knappen for Reisehjelp øvst i visninga.\n\nVi jobbar fortsatt med funksjonen, og ny funksjonalitet vil komme fortløpande. Del gjerne tilbakemeldinga di!',
```

**Review by rosvik — APPROVED** [review]
One final round of nynorsk nitpicks 😅 But otherwise, this looks good 🚀

**Review by gorandalum — APPROVED** [review]
